Warn when ignore-tidy-linelength is present, but no lines are too long

varkor opened the pull request:
It's easy for a `// ignore-tidy-linelength` to be added when there is a genuine need to ignore a file's line length, but then after refactoring the need is gone, but the tidy directive is not removed. This means that in the future, further editing may accidentally add unnecessarily long lines. This change forces `// ignore-tidy-linelength` to be used exactly when necessary, to make sure such changes are intentional.

rust-highfive commented:
r? @nikomatsakis

(rust_highfive has picked a reviewer for you, use r? to override)

varkor commented:
r? @kennytm

kennytm commented on src/test/ui/borrowck/borrowck-unary-move.rs:
```diff
@@ -1,3 +1,6 @@
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
```
Is this change intended?

varkor replied:
Whoops, no, I must have messed up the rebase.

kennytm commented on src/test/ui/cannot-mutate-captured-non-mut-var.rs:
```diff
@@ -1,3 +1,6 @@
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
```
(Ditto)

kennytm commented on src/test/ui/regions/region-borrow-params-issue-29793-big.rs:
```diff
@@ -6,6 +6,9 @@
 // able to be invoked).
 
+// revisions: ast mir
+//[mir]compile-flags: -Z borrowck=mir
```
(Ditto)

kennytm commented:
r=me after Travis passes.

petrochenkov commented:
This kind of logic is applicable to all tidy rules.

`// ignore-tidy-x` && `not x` -> tidy error

bors commented:
:umbrella: The latest upstream changes (presumably #60168) made this pull request unmergeable. Please resolve the merge conflicts.

varkor commented:
@kennytm: I've updated the PR to also warn when other tidy ignore directives are present, but have no effect. Could you check to make sure the changes are still okay?

kennytm commented:
@bors r+

bors commented:
:pushpin: Commit c2895a54087cc10a09d42d52beccf1fdd569be2a has been approved by `kennytm`

<!-- @bors r=kennytm c2895a54087cc10a09d42d52beccf1fdd569be2a -->

kennytm commented:
@bors r-

This PR failed in the rollup #60189, because #60172 has accidentally removed the CRLF line endings from [src/test/run-pass/issues/issue-16278.rs](https://github.com/rust-lang/rust/blob/0f11354a9c1bf0c5ac250c7fa2bafc289a662f42/src/test/run-pass/issues/issue-16278.rs) (see #16278). They need to be added back.

<img width="410" alt="Screenshot 2019-04-23 at 17 50 26-fs8" src="https://user-images.githubusercontent.com/103023/56571969-9fdb0c00-65f0-11e9-84c7-593a584876eb.png">

varkor commented:
> This PR failed in the rollup #60189, because #60172 has accidentally removed the CRLF line endings 

Ah, this was very easy to miss — the line endings were converted silently as soon as I modified the file 😬
I've reverted the change and am checking everything now passes again.

varkor commented:
@bors r=kennytm

bors commented:
:pushpin: Commit 5ab5806c875b9336cd97ff5e58f803b6c95fb9b2 has been approved by `kennytm`

<!-- @bors r=kennytm 5ab5806c875b9336cd97ff5e58f803b6c95fb9b2 -->
